File: src/theme/typography.ts
/**
 * Zutsu-Log タイポグラフィ
 * フォントファミリー、サイズ、ウェイトの定義
 * ハードコード禁止 - 必ずこのファイルから参照すること
 */

export const fontFamily = {
  // React Nativeではシステムフォントを使用
  // iOS: San Francisco, Android: Roboto がデフォルト
  // 日本語フォントは expo-google-fonts で追加可能
  sans: 'System',
  // カスタムフォント読み込み後に使用
  notoSansJP: 'NotoSansJP',
  inter: 'Inter',
} as const;

export const fontSize = {
  /** 10px - ラベル、バッジ、キャプション */
  xs: 10,
  /** 12px - サブテキスト、補足 */
  sm: 12,
  /** 13px - 小さめの本文 */
  md: 13,
  /** 14px - 本文 */
  base: 14,
  /** 15px - やや大きめの本文 */
  normal: 15,
  /** 16px - 小見出し */
  lg: 16,
  /** 17px - ヘッダータイトル */
  title: 17,
  /** 18px - セクションタイトル */
  xl: 18,
  /** 24px - 大きな数値 */
  '2xl': 24,
  /** 30px - ヒーロー数値 */
  '3xl': 30,
} as const;

export const fontWeight = {
  /** 通常 */
  normal: '400' as const,
  /** 中間 */
  medium: '500' as const,
  /** 半太字 */
  semibold: '600' as const,
  /** 太字 */
  bold: '700' as const,
} as const;

export const lineHeight = {
  /** タイト - 見出し用 */
  tight: 1.2,
  /** 標準 - 短いテキスト用 */
  normal: 1.4,
  /** ゆったり - 本文用 */
  relaxed: 1.6,
} as const;

export const letterSpacing = {
  /** タイト */
  tighter: -0.5,
  tight: -0.25,
  /** 標準 */
  normal: 0,
  /** 広め - ラベル用 */
  wide: 0.5,
  /** かなり広め - 大文字ラベル用 */
  wider: 1,
  /** 最も広い */
  widest: 2,
} as const;

// よく使うテキストスタイルのプリセット
export const textStyles = {
  // ヘッダータイトル
  headerTitle: {
    fontSize: fontSize.title,
    fontWeight: fontWeight.bold,
    letterSpacing: letterSpacing.tight,
  },
  // セクションタイトル
  sectionTitle: {
    fontSize: fontSize.lg,
    fontWeight: fontWeight.bold,
    lineHeight: lineHeight.tight,
  },
  // 大きな数値（気圧など）
  heroNumber: {
    fontSize: fontSize['3xl'],
    fontWeight: fontWeight.bold,
    lineHeight: lineHeight.tight,
  },
  // カード内の数値
  cardNumber: {
    fontSize: fontSize['2xl'],
    fontWeight: fontWeight.bold,
  },
  // 本文
  body: {
    fontSize: fontSize.base,
    fontWeight: fontWeight.normal,
    lineHeight: lineHeight.relaxed,
  },
  // サブテキスト
  subtext: {
    fontSize: fontSize.sm,
    fontWeight: fontWeight.normal,
    lineHeight: lineHeight.normal,
  },
  // ラベル（大文字）
  label: {
    fontSize: fontSize.xs,
    fontWeight: fontWeight.bold,
    letterSpacing: letterSpacing.wider,
    textTransform: 'uppercase' as const,
  },
  // バッジテキスト
  badge: {
    fontSize: fontSize.xs,
    fontWeight: fontWeight.bold,
    letterSpacing: letterSpacing.wide,
  },
  // ボタンテキスト
  button: {
    fontSize: fontSize.md,
    fontWeight: fontWeight.bold,
  },
  // タブナビラベル
  tabLabel: {
    fontSize: fontSize.xs,
    fontWeight: fontWeight.bold,
  },
} as const;

export type FontSize = typeof fontSize;
export type FontWeight = typeof fontWeight;

File: flutter/lib/theme/typography.dart
import 'package:flutter/material.dart';
import 'app_colors.dart';

/// フォントサイズ定数
class FontSizes {
  FontSizes._();

  /// 10px - ラベル、バッジ、キャプション
  static const double xs = 10;

  /// 12px - サブテキスト、補足
  static const double sm = 12;

  /// 13px - 小さめの本文
  static const double md = 13;

  /// 14px - 本文
  static const double base = 14;

  /// 15px - やや大きめの本文
  static const double normal = 15;

  /// 16px - 小見出し
  static const double lg = 16;

  /// 17px - ヘッダータイトル
  static const double title = 17;

  /// 18px - セクションタイトル
  static const double xl = 18;

  /// 24px - 大きな数値
  static const double xxl = 24;

  /// 30px - ヒーロー数値
  static const double xxxl = 30;
}

/// 行の高さ定数
class LineHeights {
  LineHeights._();

  /// タイト - 見出し用
  static const double tight = 1.2;

  /// 標準 - 短いテキスト用
  static const double normal = 1.4;

  /// ゆったり - 本文用
  static const double relaxed = 1.6;
}

/// 文字間隔定数
class LetterSpacings {
  LetterSpacings._();

  static const double tighter = -0.5;
  static const double tight = -0.25;
  static const double normal = 0;
  static const double wide = 0.5;
  static const double wider = 1;
  static const double widest = 2;
}

/// よく使うテキストスタイルのプリセット
class AppTextStyles {
  AppTextStyles._();

  /// ヘッダータイトル
  static const TextStyle headerTitle = TextStyle(
    fontSize: FontSizes.title,
    fontWeight: FontWeight.bold,
    letterSpacing: LetterSpacings.tight,
    color: AppColors.textDark,
  );

  /// セクションタイトル
  static const TextStyle sectionTitle = TextStyle(
    fontSize: FontSizes.lg,
    fontWeight: FontWeight.bold,
    height: LineHeights.tight,
    color: AppColors.textDark,
  );

  /// 大きな数値（気圧など）
  static const TextStyle heroNumber = TextStyle(
    fontSize: FontSizes.xxxl,
    fontWeight: FontWeight.bold,
    height: LineHeights.tight,
    color: AppColors.textDark,
  );

  /// カード内の数値
  static const TextStyle cardNumber = TextStyle(
    fontSize: FontSizes.xxl,
    fontWeight: FontWeight.bold,
    color: AppColors.textDark,
  );

  /// 本文
  static const TextStyle body = TextStyle(
    fontSize: FontSizes.base,
    fontWeight: FontWeight.normal,
    height: LineHeights.relaxed,
    color: AppColors.textMain,
  );

  /// サブテキスト
  static const TextStyle subtext = TextStyle(
    fontSize: FontSizes.sm,
    fontWeight: FontWeight.normal,
    height: LineHeights.normal,
    color: AppColors.textSub,
  );

  /// ラベル
  static const TextStyle label = TextStyle(
    fontSize: FontSizes.xs,
    fontWeight: FontWeight.bold,
    letterSpacing: LetterSpacings.wider,
    color: AppColors.textMuted,
  );

  /// バッジテキスト
  static const TextStyle badge = TextStyle(
    fontSize: FontSizes.xs,
    fontWeight: FontWeight.bold,
    letterSpacing: LetterSpacings.wide,
  );

  /// ボタンテキスト
  static const TextStyle button = TextStyle(
    fontSize: FontSizes.md,
    fontWeight: FontWeight.bold,
  );

  /// タブナビラベル
  static const TextStyle tabLabel = TextStyle(
    fontSize: FontSizes.xs,
    fontWeight: FontWeight.bold,
  );

  /// プライマリカラーのテキスト
  static const TextStyle primaryText = TextStyle(
    fontSize: FontSizes.base,
    fontWeight: FontWeight.w500,
    color: AppColors.primary,
  );

  /// ミュートテキスト
  static const TextStyle mutedText = TextStyle(
    fontSize: FontSizes.sm,
    fontWeight: FontWeight.normal,
    color: AppColors.textMuted,
  );
}
